optical/converter/yolo.py

```diff
@@ -1,17 +1,136 @@
 
-from .base import FormatSpec
-from typing import Union
+
 import os
+import imagesize
+from pathlib import Path
+import yaml
+import warnings
+from typing import Union
+import numpy as np
+import pandas as pd
+from .base import FormatSpec
+from .utils import exists, get_image_dir, get_annotation_dir
 
 
 class Yolo(FormatSpec):
     def __init__(self, root: Union[str, os.PathLike]):
-        pass
+        self.root = root
+        self.class_file = Path(os.path.join(self.root, "data.yaml"))
+        self._image_dir = get_image_dir(root)
+        self._annotation_dir = get_annotation_dir(root)
+        self._has_image_split = False
+        assert exists(self._image_dir), "root is missing 'images' directory."
+        assert exists(self._annotation_dir), "root is missing 'annotations' directory."
+        self._splits = self._find_splits()
+        self._resolve_dataframe()
+
+    def _find_splits(self):
+        im_splits = [x.name for x in Path(self._image_dir).iterdir() if x.is_dir()]
+        ann_splits = [x.name for x in Path(self._annotation_dir).iterdir() if x.is_dir()]
+
+        if im_splits:
+            self._has_image_split = True
+
+        no_anns = set(im_splits).difference(ann_splits)
+        if no_anns:
+            warnings.warn(f"no annotation found for {','.join(list(no_anns))}")
+        return ann_splits
 
     def _resolve_dataframe(self):
-        return super()._resolve_dataframe()
+        img_names = []
+        cls_ids = []
+        x_center = []
+        y_center = []
+        box_width = []
+        box_height = []
+        splits = []
+        image_height = []
+        image_width = []
+        names_category = []
+        for split in self._splits:
+            ann_dir_files = os.path.join(self._annotation_dir, split)
+            img_dir_files = os.path.join(self._image_dir, split)
+            txt_files = [x for x in Path(ann_dir_files).glob("*.txt")]
+            img_files = [x for x in Path(img_dir_files).glob("*.jpg")]
+            for txt, img in zip(txt_files, img_files):
+                file_names = os.path.basename(img)
+                image_widths, image_heights = imagesize.get(img)
+                image_height.append(image_heights)
+                image_width.append(image_widths)
+                with open(txt, "rt") as fd:
+                    first_line = fd.readline()
+                    splited = first_line.split()
+                    class_id = splited[0]
+                    x_cent = splited[1]
+                    y_cent = splited[2]
+                    box_widths = splited[3]
+                    box_heights = splited[4]
+                    img_names.append(file_names)
+                    x_center.append(x_cent)
+                    y_center.append(y_cent)
+                    box_width.append(box_widths)
+                    box_height.append(box_heights)
+                    cls_ids.append(class_id)
+                    splits.append(split)
+
+        if self.class_file.is_file():
```

Comment:
Better to move this snippet to a separate method, so that it is modularized.

Reply:
also more concise:
```
class_id, x_cent, y_cent, ... = first_line.split()
```